tests/unit/core/statistics/test_constraints.py

```diff
@@ -460,11 +462,128 @@ def test_serialization_deserialization_values_in_set_constraint():
 
     TEST_LOGGER.info(f"Serialize columnValuesInSetConstraint from deserialized representation:\n {cvisc.to_protobuf()}")
 
-    assert json_value["name"] == f"value {Op.Name(Op.IN_SET)} " + str(val_set)
-    assert json_value["op"] == Op.Name(Op.IN_SET)
+    assert json_value["name"] == f"value {Op.Name(Op.IN)} " + str(val_set)
+    assert json_value["op"] == Op.Name(Op.IN)
     assert json_value["valueSet"][0] == list(val_set)
 
 
 def test_column_values_in_set_wrong_datatype():
     with pytest.raises(TypeError):
         cvisc = columnValuesInSetConstraint(value_set=1)
+
+
+def test_column_values_type_equals_constraint_apply(local_config_path, df_lending_club):
+    cvtc = columnValuesTypeEqualsConstraint(expected_type=InferredType.Type.FRACTIONAL)
+    dc = DatasetConstraints(None, summary_constraints={"annual_inc": [cvtc]})
+    config = load_config(local_config_path)
+    session = session_from_config(config)
+    profile = session.log_dataframe(df_lending_club, "test.data", constraints=dc)
+    session.close()
+    report = profile.apply_summary_constraints()
+
+    assert report[0][1][0][0] == f"summary column_values_type {Op.Name(Op.EQ)} {InferredType.Type.Name(InferredType.Type.FRACTIONAL)}"
+    assert report[0][1][0][1] == 1
+    assert report[0][1][0][2] == 0
+
+
+def test_merge_column_values_type_equals_constraint_different_values():
+    c1 = columnValuesTypeEqualsConstraint(expected_type=InferredType.Type.FRACTIONAL)
+    c2 = columnValuesTypeEqualsConstraint(expected_type=InferredType.Type.NULL)
+    with pytest.raises(AssertionError):
+        c1.merge(c2)
```

Comment:
This looks like it might be better as an exception rather than an `AssertionError`, but it also looks like we do this in a few places throughout the code base. Probably nothing to change here for this PR, but check with @jamie256 for designing the next version of whylogs...